applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,242 +60,273 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
+     * @param key The key to delete.
+     * @return the deleted ConfigurationSetting or none if didn't exist.
      */
-    private Flux<RestPagedResponse<ConfigurationSetting>> listSinglePageAsync(KeyValueListFilter filter) {
-        Mono<RestResponse<PageImpl<ConfigurationSetting>>> result;
-        if (filter != null) {
-            result = service.listKeyValues(baseUri.toString(), filter.key(), filter.label(), filter.fields(), filter.acceptDateTime(), filter.range());
-        } else {
-            result = service.listKeyValues(baseUri.toString(), null, null, null, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key) {
+        if (key == null || key.isEmpty()) {
+            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
         }
 
-        return result.flatMapMany(p ->
-                Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(),
-                        p.headers(), p.statusCode())));
+        return deleteKeyValue(key, null, null);
     }
 
     /**
-     * Gets all ConfigurationSetting settings.
+     * Deletes the ConfigurationSetting.
      *
-     * @param nextPageLink The NextLink from the previous successful call to List operation.
-     * @return the observable to the Page&lt;ConfigurationSetting&gt; object.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @param key   key of the keyValue to delete
+     * @param label Optional. If not specified, {@link ConfigurationSetting#NULL_LABEL} is used.
+     * @param etag  Optional. If specified, will only delete the key if its current etag matches. (ie. No one has
+     *              changed the value yet.)
+     * @return the deleted ConfigurationSetting or none if didn't exist.
      */
-    private Flux<RestPagedResponse<ConfigurationSetting>> listNextAsync(@NonNull String nextPageLink) {
-        return listNextSinglePageAsync(nextPageLink)
-                       .concatMap(page -> {
-                           String nextPageLink1 = page.nextLink();
-                           if (nextPageLink1 == null) {
-                               return Flux.just(page);
-                           }
-                           return Flux.just(page).concatWith(p -> listNextAsync(nextPageLink1));
-                       });
-    }
-
-    private Flux<RestPagedResponse<ConfigurationSetting>> listNextSinglePageAsync(@NonNull String nextPageLink) {
-        if (nextPageLink == null) {
-            throw new IllegalArgumentException("Parameter nextPageLink is required and cannot be null.");
+    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key, String label, String etag) {
+        if (key == null || key.isEmpty()) {
+            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
```

Comment:
I always like to be a bit clearer when referring to parameter names - I'd suggest "Parameter 'key' is ..." just to make it a bit clearer.